internal/dag/builder.go

```diff
@@ -989,9 +989,9 @@ func (b *Builder) processHTTPProxyTCPProxy(sw *ObjectStatusWriter, httpproxy *pr
 		var proxy TCPProxy
 		for _, service := range httpproxy.Spec.TCPProxy.Services {
 			m := k8s.FullName{Name: service.Name, Namespace: httpproxy.Namespace}
-			s := b.lookupService(m, intstr.FromInt(service.Port))
-			if s == nil {
-				sw.SetInvalid("tcpproxy: service %s/%s/%d: not found", httpproxy.Namespace, service.Name, service.Port)
+			s, err := b.lookupService(m, intstr.FromInt(service.Port))
+			if err != nil {
+				sw.SetInvalid("tcpproxy: service %s/%s/%d: %s", httpproxy.Namespace, service.Name, service.Port, err)
```

Comment:
Can we take the opportunity to align the status error formatting? Current best practice seems to be starting with the field name and quoting any resource names

So maybe it could be something like this:
```Go
sw.SetInvalid("Spec.TCPProxy unresolved service reference: %s", err)
```

And something equivalent for the other lookup